src/components/logger/logger.js

```diff
@@ -53,26 +55,26 @@ export default class Logger extends React.Component {
   };
 
   state = {
+    logs: [],
     events: [],
   };
 
   onClear = () => this.setState({ events: [] });
+  onEvent = (event) => this.setState((state) => ({ events: [...state.events, event] }));
 
-  componentWillUpdate({ logs }) {
-    this.logsSubscriptions.map((sub) => sub.unsubscribe());
-    this.logsSubscriptions = logs.map((obs) => {
-      return obs.subscribe((event) => {
-        this.setState((state) => ({ events: [...state.events, event] }));
-      });
-    });
+  componentWillUpdate(nextProps) {
+    const { logs: prevLogs } = this.state;
+    const { logs: nextLogs } = nextProps;
+
+    if (!isEqual(prevLogs, nextLogs)) {
+      this.sub.unsubscribe();
+      this.sub = merge(...nextLogs).subscribe(this.onEvent);
```

Comment:
@95kach  Why to unsubscribe then merge then subscribe again ?

Reply:
Because for example the first time we rendered Logger with [logs: ServiceA.method$()]

Then in the runtime, we updated the logs to: [logs: ServiceA.method$(), ServiceB.method$()]

Etc. and each the time when we are changing the props of logs we should update log events.

For example, I'm getting to show in logger only ServiceA events, then in the next moment, I decided to show ServiceB events, why we should keep old ServiceA events?

Reply:
Logs it's an Observable, I can't just replace it. I need to close the old opened subscriptions. Before update the component with the new logs.

Reply:
And why merge? Because for me keep one sub instead of the array of subs is easier

Reply:
ok